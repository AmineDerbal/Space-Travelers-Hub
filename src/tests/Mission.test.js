import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import {
  render,
  screen, fireEvent,
} from '@testing-library/react';
import configureMockStore from 'redux-mock-store';
import store from '../redux/store';
import Mission from '../Components/Mission';

it('renders Missions correctly', () => {
  const tree = renderer.create(
    <React.StrictMode>
      <Provider store={store}>
        <Router>
          <Mission />
        </Router>
      </Provider>
    </React.StrictMode>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('join a mission', () => {
  const mockStore = configureMockStore();
  const initialState = {
    missions: {
      missions: [
        {
          id: '1',
          name: 'Thaicom',
          description: 'Thaicom is the name of a series of communications satellites operated from Thailand, and also the name of Thaicom Public Company Limited, which is the company that owns and operates the Thaicom satellite fleet and other telecommunication businesses in Thailand and throughout the Asia-Pacific region. The satellite projects were named Thaicom by the King of Thailand, His Majesty the King Bhumibol Adulyadej, as a symbol of the linkage between Thailand and modern communications technology.',
        },
      ],
      isLoading: false,
    },

  };

  const store = mockStore(initialState);
  store.clearActions();
  render(
    <Provider store={store}>
      <Mission />
    </Provider>,
  );

  const joinButton = screen.getByRole('button', { name: /Join Mission/i });
  fireEvent.click(joinButton);

  expect(store.getActions()).toEqual([{ type: 'missions/joinMission' }]);
});
