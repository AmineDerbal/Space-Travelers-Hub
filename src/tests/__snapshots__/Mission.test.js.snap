// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Missions correctly 1`] = `
<tr>
  <th
    className="heading"
    scope="row"
  />
  <td
    className="heading col-lg-7"
  />
  <td
    className="btns"
  >
    <span
      className="badge text-bg-secondary"
    >
      Not A Member
    </span>
  </td>
  <td
    className="btns"
  >
    <button
      className="btn btn-outline-secondary test-small btn-sm"
      onClick={[Function]}
      type="button"
    >
      Join Mission
    </button>
  </td>
</tr>
`;
